### ISSUE TITLE
Introducing `'use client'` to angular

### ISSUE THREAD

**Issue by kai1992cool** [angular/angular#57422]
### Description

Next.js has this feature called `use client` where it implies the component will not be rendered on the server, this is very useful for generic wrapper components for libraries that do not support SSR and have no intention to support SSR. If we introduce this, it will be the least possible code implementation instead of `@defer`, `isPlatformBrowser`.

### Proposed solution

Make the angular compiler/build recognize `use client` and stop rendering them on SSR

### Alternatives considered

This would be a good to have feature, although there are other ways to solve this.

**Comment by alxhub** [comment]
`use client` is a thing in Next.js because certain features assume components are server components (RSC). `use client` is then used to declare the client/server boundary. Since Angular doesn't have the concept of server-only components, I don't think this feature makes sense for us.

What are some reasons / use cases why a component couldn't be made renderable on the server in the first place?

**Comment by JeanMeche** [comment]
I've been wondering that myself, could the framework provide some kind of possibilities to declare client only components. 

Imagine having an SSR app, that displayed charts. Those charts would be canvas based (like many libs do). It doesn't make sense to run them on the server.

Could we imagine an option (in the `@Component` decorator ?) that would indicate to the framework, do not render that component on the server, let the client do it. 

You could see it a bit like the `ngSkipHydration` attribute, but on the component directly (like `hostDirectives`). Only this would skip entirely the rendering on the server. 

What do you think of such feature. ?

**Comment by kai1992cool** [comment]
@eneajaho  that has the problem that I have to specify it on each implementation of the component that should be browser side only, if we can keep it at the component level that would be the easiest implementation

**Comment by alxhub** [comment]
Generally in Angular though, you'd initialize those charts inside `afterNextRender` which already doesn't run on the server. In that sense, you _want_ the chart component to render on the server and create whatever template/scaffolding is needed, but stop short of trying to initialize the charting library. You might even want to render something different (e.g. a shimmer) while you wait for hydration.

**Comment by splincode** [comment]
```html
<header> Header </header>

@clientOnly {
  <wysiwyg />
}

```

**Comment by alxhub** [comment]
Closing as Angular has a different model than React for "client-only" behavior, and already has ways of expressing this in component code.

**Comment by kai1992cool** [comment]
The use case I want to show the need for a component to be declared as `client side rendering only`.

Take this example of `ng-apex-charts`

[I would like to use apexcharts and ng-apexcharts in angular 18 with SSR](https://stackoverflow.com/questions/78600483/i-would-like-to-use-apexcharts-and-ng-apexcharts-in-angular-18-with-ssr)

Packages that do not support SSR. can only be used if we create a wrapper component with the import on the standalone component, followed by a wrapper component with it inside the defer block.

I created a component `apx-chart.component.ts` which contains the module `NgApexchartsModule` in the imports array. This will serve as the wrapper component for the apx chart library component with the same `@Input` and `@Output`.

Then I had to create a wrapper component, which is `@defer` wrapped on the `apx-chart.component.ts` on the component  `apx-chart-ssr.component.ts`.

[Stackblitz Demo -> cd test -> npm i -> npm run start](https://stackblitz.com/edit/stackblitz-starters-npudzo?file=test%2Fsrc%2Fapp%2Fapx-chart-ssr%2Fapx-chart-ssr.component.ts)

---

So looking at this example, there are few packages that do not support SSR but are usable on SSR, using this wrapper method.

As a solution for this, I propose on the component decorator we add a property `clientOnly` so that the component renders only on the client and we are to use non-ssr supported packages with ease, without the problem of multiple `@defer` blocks.

`@clientOnly` solution is just a remake of `@defer` so there is less impact in creating this.

The best way is to have the logic set at the component decorator.

```
@Component({
  selector: 'app-apx-chart-ssr',
  standalone: true,
  clientOnly: true,
  imports: [ApxChartComponent],
  templateUrl: './apx-chart-ssr.component.html',
})
export class ApxChartSsrComponent {
   ...
```

FYI @alxhub @JeanMeche
